src/main/azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto

```diff
@@ -225,31 +309,50 @@ message RpcLog {
     None = 6;
   }
 
+  // Unique id for invocation (if exists)
   string invocation_id = 1;
+
+  // TOD: This should be an enum
+  // Category for the log (startup, load, invocation, etc.)
   string category = 2;
+
+  // Level for the given log message
   Level level = 3;
+
+  // Message for the given log
   string message = 4;
+
+  // Id for the even associated with this log (if exists)
   string event_id = 5;
+
+  // Exception (if exists)
   RpcException exception = 6;
 
   // json serialized property bag, or could use a type scheme like map<string, TypedData>
   string properties = 7;
 }
 
+// Encapsulates an Exception 
 message RpcException {
+  // Source of the exception
   string source = 3;
+
+  // Stack trace for the exception
   string stack_trace = 1;
+
+  // Textual message describing hte exception
   string message = 2;
 }
 
-// TODO - solidify this
+// TODO - solidify this or remove it
 message RpcHttp {
   string method = 1;
   string url = 2; 
-  map<string,string> headers = 3;
+  map<string,TypedData> headers = 3;
```

Comment:
Although it is in another repository, but is it reasonable to have a strongly-typed HTTP header? Because by definition, all HTTP headers are String-String pairs.

Reply:
Here is a good example and discussion on valid values for http headers: [using-json-string-in-the-http-header](https://stackoverflow.com/questions/9779860/using-json-string-in-the-http-header)

Reply:
@JunyiYi - I think you are right. If the requirement for headers to be <string, string>, we should not be changing RpcHttp in proto file. Will get back to you on this.